Microsoft.Data.Analysis.Interactive/DataFrameKernelExtension.cs

```diff
@@ -0,0 +1,131 @@
+
+                    var paginateScriptPrevTen = BuildHideRowsScript(uniqueId) + UpdatePageIndex(uniqueId, -10, (maxRows - 1) / SIZE) + BuildPageScript(uniqueId, SIZE);
+                    footer.Add(button[style: "margin: 2px;", onclick: paginateScriptPrevTen]("⏪"));
+
+                    var paginateScriptPrev = BuildHideRowsScript(uniqueId) + UpdatePageIndex(uniqueId, -1, (maxRows - 1) / SIZE) + BuildPageScript(uniqueId, SIZE);
+                    footer.Add(button[style: "margin: 2px;", onclick: paginateScriptPrev]("◀️"));
+
+                    footer.Add(b[style: "margin: 2px;"]("Page"));
+                    footer.Add(b[id: $"page_{uniqueId}", style: "margin: 2px;"]("1"));
+
+                    var paginateScriptNext = BuildHideRowsScript(uniqueId) + UpdatePageIndex(uniqueId, 1, (maxRows - 1) / SIZE) + BuildPageScript(uniqueId, SIZE);
+                    footer.Add(button[style: "margin: 2px;", onclick: paginateScriptNext]("▶️"));
+
+                    var paginateScriptNextTen = BuildHideRowsScript(uniqueId) + UpdatePageIndex(uniqueId, 10, (maxRows - 1) / SIZE) + BuildPageScript(uniqueId, SIZE);
+                    footer.Add(button[style: "margin: 2px;", onclick: paginateScriptNextTen]("⏩"));
+
+                    var paginateScriptLast = BuildHideRowsScript(uniqueId) + GotoPageIndex(uniqueId, (maxRows - 1) / SIZE) + BuildPageScript(uniqueId, SIZE);
+                    footer.Add(button[style: "margin: 2px;", onclick: paginateScriptLast]("⏭️"));
+                }
+                else
+                {
+                    BuildHideRowsScript(uniqueId);
+                    footer.Add(b[style: "margin: 2px;"]("Page"));
+                    footer.Add(b[id: $"page_{uniqueId}", style: "margin: 2px;"]("0"));
+                }
+
+                //table
+                var t = table[id: $"table_{uniqueId}"](
+                    caption(title),
+                    thead(tr(header)),
+                    tbody(rows.Select(r => tr[style: "display: none"](r))),
+                    tfoot(tr(td[colspan: df.Columns.Count + 1, style: "text-align: center;"](footer)))
+                );
+                writer.Write(t);
+
+                //show first page
+                writer.Write($"<script>{BuildPageScript(uniqueId, SIZE)}</script>");
+
+            }, "text/html");
+
+            return Task.CompletedTask;
+
+
+            static string BuildHideRowsScript(long uniqueId)
+            {
+                var script = $"var allRows = document.querySelectorAll('#table_{uniqueId} tbody tr:nth-child(n)'); ";
+                script += "for (var i = 0; i < allRows.length; i++) { allRows[i].style.display='none'; } ";
+                return script;
+            }
+
+            static string BuildPageScript(long uniqueId, int size)
+            {
+                var script = $"var page = parseInt(document.querySelector('#page_{uniqueId}').innerHTML) - 1; ";
```

Comment:
swap all `var` with `let`, this will make this code safer

Reply:
here I need "page" variable to be reusable, but I see your point.